src/components/AssetTree/AssetTree.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react';
+import { Tree } from 'antd';
+import { AntTreeNode } from 'antd/lib/tree';
+import {
+  AssetTreeType,
+  TreeNodeType,
+  AssetType,
+  OnSelectReturnType,
+  TreeNodeData,
+} from 'utils/validators';
+
+const { TreeNode } = Tree;
+
+class AssetTree extends Component<AssetTreeType> {
+  state = {
+    treeData: [],
+    expandedKeys: {},
+  };
+
+  async componentDidMount() {
+    const { assets, defaultExpandedKeys } = this.props;
+
+    if (assets && assets.length > 0) {
+      this.setState({
+        treeData: this.mapDataAssets(assets),
+        expandedKeys: defaultExpandedKeys
+          ? this.toKeys(defaultExpandedKeys)
+          : {},
+      });
+    }
+  }
+
+  mapDataAssets = (assets: AssetType[]) => {
+    const nodes: { [name: string]: TreeNodeData } = {};
+
+    assets.forEach(asset => {
+      nodes[asset.id] = this.returnPretty(asset);
+    });
+
+    const addedAsChildren: (number | string)[] = [];
+
+    assets.forEach(asset => {
+      const { parentId } = asset;
+      const node = nodes[parentId as number]; // casting is not a problem. It will return undefined if not found
+      if (!node) {
+        return;
+      }
+      if (node.children) {
+        node.children.push(nodes[asset.id]);
+      } else {
+        node.children = [nodes[asset.id]];
+      }
+      addedAsChildren.push(asset.id);
+      node.isLeaf = false;
+    });
+
+    addedAsChildren.forEach(id => {
+      delete nodes[id];
+    });
+
+    return Object.keys(nodes).map(id => {
+      return nodes[id];
+    });
+  };
+
+  returnPretty = (asset: AssetType) => ({
+    title: `${asset.name}: ${asset.description}`,
+    key: asset.id,
+    node: asset,
+    isLeaf: true,
+  });
+
+  onLoadData = async (treeNode: AntTreeNode) => {
+    const { loadData } = this.props;
+
+    if (treeNode.props.children) {
+      return;
+    }
+    const assetId = treeNode.props.eventKey;
+
+    if (loadData && assetId) {
+      const query = {
```

Comment:
Here is something we can improve over time. Can we find a smarter query than alway ask for depth=2? What if the result for depth=2 is greater than 1000? Then we loose some data.
The API / SDK isn't perfect here but something to keep in mind to later.